test/regression/issue314.jl

```diff
@@ -0,0 +1,2 @@
+@test JSON.parse(IOBuffer("123")) == 123
```

Comment:
No need to create a new file for this, just add it to e.g. https://github.com/JuliaIO/JSON.jl/blob/1c24980c320842e17720c173be767c161f36feb9/test/runtests.jl#L36